sysl2/sysl/ints.go

```diff
@@ -0,0 +1,303 @@
+		diagramTitle = appAttrs["title"].GetS()
+		//TODO(Jack) fmt title with appAttrs["title"]
+	} else {
+		//TODO(Jack) fmt title with args.title
+	}
+
+	var sb strings.Builder
+	if args.epa || endptAttrs["view"].GetS() == "epa" {
+		//TODO generate state view
+		fmt.Fprintln(&sb, "''''''''''''''''''''''''''''''''''''''''''")
+		fmt.Fprintln(&sb, "''                                      ''")
+		fmt.Fprintln(&sb, "''  AUTOGENERATED CODE -- DO NOT EDIT!  ''")
+		fmt.Fprintln(&sb, "''                                      ''")
+		fmt.Fprintln(&sb, "''''''''''''''''''''''''''''''''''''''''''")
+		fmt.Fprintln(&sb)
+
+		fmt.Fprintln(&sb, "@startuml")
+		if diagramTitle != "" {
+			fmt.Fprintln(&sb, "title "+diagramTitle)
+		}
+		fmt.Fprintln(&sb, "left to right direction")
+		fmt.Fprintln(&sb, "scale max 16384 height")
+		fmt.Fprintln(&sb, "hide empty description")
+		fmt.Fprintln(&sb, "skinparam state {")
+		fmt.Fprintln(&sb, "  BackgroundColor FloralWhite")
+		fmt.Fprintln(&sb, "  BorderColor Black")
+		fmt.Fprintln(&sb, "  ArrowColor Crimson")
+		if highLightColor != "" {
+			fmt.Fprintln(&sb, "  BackgroundColor<<highlight>> "+highLightColor)
+		}
+		if arrowColor != "" {
+			fmt.Fprintln(&sb, "  ArrowColor "+arrowColor)
+		}
+
+		if indirectArrowColor != "" && indirectArrowColor != "none" {
+			fmt.Fprintln(&sb, "  ArrowColor<<indirect>> "+indirectArrowColor)
+			fmt.Fprintln(&sb, "  ArrowColor<<internal>> "+indirectArrowColor)
+		}
+		fmt.Fprintln(&sb, "}")
+		for _, v := range params.integrations {
+			if restrictBy != "" && !stringInSlice(restrictBy, getMapKeys(mod.Apps[v.Self.Name].Attrs)) && !stringInSlice(restrictBy, getMapKeys(mod.Apps[v.Target.Name].Attrs)) {
+				continue
+			}
+		}
+
+	} else {
+		fmt.Fprintln(&sb, "''''''''''''''''''''''''''''''''''''''''''")
+		fmt.Fprintln(&sb, "''                                      ''")
+		fmt.Fprintln(&sb, "''  AUTOGENERATED CODE -- DO NOT EDIT!  ''")
+		fmt.Fprintln(&sb, "''                                      ''")
+		fmt.Fprintln(&sb, "''''''''''''''''''''''''''''''''''''''''''")
+		fmt.Fprintln(&sb)
+
+		fmt.Fprintln(&sb, "@startuml")
+		fmt.Fprintln(&sb, "hide stereotype")
+		fmt.Fprintln(&sb, "scale max 16384 height")
+		fmt.Fprintln(&sb, "skinparam component {")
+		fmt.Fprintln(&sb, "  BackgroundColor FloralWhite")
+		fmt.Fprintln(&sb, "  BorderColor Black")
+		fmt.Fprintln(&sb, "  ArrowColor Crimson")
+		if highLightColor != "" {
+			fmt.Fprintln(&sb, "  BackgroundColor<<highlight>> "+highLightColor)
+		}
+		if arrowColor != "" {
+			fmt.Fprintln(&sb, "  ArrowColor "+arrowColor)
+		}
+
+		if indirectArrowColor != "" && indirectArrowColor != "none" {
+			fmt.Fprintln(&sb, "  ArrowColor<<indirect>> "+indirectArrowColor)
+		}
+		fmt.Fprintln(&sb, "}")
+
+		if args.clustered || endptAttrs["view"].GetS() == "clustered" {
+			clusters := make(map[string][]string)
+			for _, v := range params.apps {
+				cluster := strings.Split(v, " :: ")
+				if len(cluster) != 0 {
```

Comment:
`len(cluster) != 0` always be true.
Refer to https://golang.org/pkg/strings/#Split
If both s and sep are empty, Split returns an empty slice.

```suggestion
      if len(cluster) > 1 {
            clusters[cluster[0]] = cluster[1:]
      }
```